### PR TITLE
POM-88 Searching material aid offers

### PR DESCRIPTION
**Pull request by sfc-gh-tkaminski**

### CODE REVIEW

File: src/main/resources/db/migration/V8__material_aid_offer.sql [hunk 1]
```diff
@@ -5,7 +5,5 @@ create table material_aid_offer (
     description     text,
     category        text,
     city            text,
-    region          text,
-
-    CONSTRAINT PK_MAO_ID PRIMARY KEY (ID)
```
Comment (sfc-gh-krzymkowski) [inline src/main/resources/db/migration/V8__material_aid_offer.sql:10]: Wasn't the primary key a good thing? Removed by accident?

File: src/main/java/pl/gov/coi/pomocua/ads/materialaid/MaterialAidResource.java [hunk 2]
```diff
@@ -26,9 +26,11 @@ public MaterialAidOffer postMaterialAidOffer(
         return repository.save(materialAidOffer);
     }
 
+    @Operation(description = "Allows to search for material aid offers using different criteria (passes as query params). Each criteria is optional.")
     @GetMapping("material-aid")
-    public Page<MaterialAidOffer> list(Pageable pageRequest) {
-        return repository.findAll(pageRequest);
+    public Page<MaterialAidOffer> list(Pageable pageRequest, MaterialAidOfferSearchCriteria searchCriteria) {
+        Page<MaterialAidOffer> all = repository.findAll(MaterialAidOfferSpecifications.from(searchCriteria), pageRequest);
```
Comment (sfc-gh-krzymkowski) [inline src/main/java/pl/gov/coi/pomocua/ads/materialaid/MaterialAidResource.java:32]: Unnecessary local var -> just `return repo.findAll(...`

Comment (sfc-gh-tkaminski) [reply src/main/java/pl/gov/coi/pomocua/ads/materialaid/MaterialAidResource.java:32]: I forgot to revert changes after debugging :-) thanks

Comment (sfc-gh-tkaminski) [reply src/main/java/pl/gov/coi/pomocua/ads/materialaid/MaterialAidResource.java]: done

File: src/main/resources/db/migration/V8__material_aid_offer.sql (hunk 1 above)
Comment (sfc-gh-tkaminski) [reply src/main/resources/db/migration/V8__material_aid_offer.sql:10]: wasn't sure if it should be here, other tables has no such declarations

Comment (sfc-gh-krzymkowski) [reply src/main/resources/db/migration/V8__material_aid_offer.sql:10]: All of them should have primary keys. But that's maybe a separate task to review that along with indexes

**Pull request merged**